### PR TITLE
Add SQL specific tests for messages, part 1

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Rewrite APIs similar to transfer tasks, timer tasks
* Add queue message tests

**Why?**
This PR provides better test coverage and code management.

**How did you test it?**
Run tests.

**Potential risks**
Minor business logic code change.

### CODE REVIEW

File: common/persistence/sql/sqlplugin/queue_messate.go [hunk 1]
```diff
@@ -26,10 +26,17 @@ package sqlplugin
 
 import (
```
Comment (alexshtin) [inline common/persistence/sql/sqlplugin/queue_messate.go:27]: Typo in file name?

File: common/persistence/sql/sqlplugin/queue_messate.go [hunk 2]
```diff
@@ -38,20 +45,35 @@ type (
...
 	}
 
-	Queue interface {
-		InsertIntoQueue(row *QueueRow) (sql.Result, error)
+	QueueMessage interface {
+		InsertIntoMessages(row []QueueRow) (sql.Result, error)
+		SelectFromMessages(filter QueueMessagesFilter) ([]QueueRow, error)
+		RangeSelectFromMessages(filter QueueMessagesRangeFilter) ([]QueueRow, error)
+		DeleteFromMessages(filter QueueMessagesFilter) (sql.Result, error)
+		RangeDeleteFromMessages(filter QueueMessagesRangeFilter) (sql.Result, error)
```
Comment (alexshtin) [inline common/persistence/sql/sqlplugin/queue_messate.go:73]: Consider:
```suggestion
		Enqueue(row []QueueRow) (sql.Result, error)
		Select(filter QueueMessagesFilter) ([]QueueRow, error)
		SelectRange(filter QueueMessagesRangeFilter) ([]QueueRow, error)
		Delete(filter QueueMessagesFilter) (sql.Result, error)
		DeleteRange(filter QueueMessagesRangeFilter) (sql.Result, error)
```

**Review by alexshtin — APPROVED** [review]

File: common/persistence/sql/sqlplugin/queue_messate.go (hunk 2 above)
Comment (wxing1292) [reply common/persistence/sql/sqlplugin/queue_messate.go:73]: sample current convention: 
https://github.com/temporalio/temporal/blob/9d5c94e/common/persistence/sql/sqlplugin/history_transfer_tasks.go#L54-L64

plus the function name should explicitly express the functionality.

Comment (alexshtin) [reply common/persistence/sql/sqlplugin/queue_messate.go]: Seems to me `FromTransferTasks` is useless there too.

Comment (alexshtin) [reply common/persistence/sql/sqlplugin/queue_messate.go]: I usually run mental test and answer to the question "Does this particular word really adds more context?" If interface name is `QueueMessage` and method accepts `QueueMessagesFilter`, is it clear what `Select` method does? Seems to me it is.

Comment (wxing1292) [reply common/persistence/sql/sqlplugin/queue_messate.go]: in my mind:
```
store1.Select
store2.Select
```
vs
```
store1.SelectFromMessage
store2.SelectFromTransferTasks
```
the latter one is more easier to read, plus the latter is the naming convention.

Comment (wxing1292) [reply common/persistence/sql/sqlplugin/queue_messate.go]: BTW, we have a #810 task before making the entire persistence layer modular

Comment (wxing1292) [reply common/persistence/sql/sqlplugin/queue_messate.go]: Plus the name / interface collision here: https://github.com/temporalio/temporal/blob/master/common/persistence/sql/sqlplugin/interfaces.go#L41

**Pull request merged**